controller/tests/test_volunteer_controller.py

```diff
@@ -0,0 +1,109 @@
+        self.assertEqual(volunteer_edited.phone, '+441111111111')
+
+    def test_edit_phone_without_international_code(self):
+        with self.assertRaises(vc.ControllerError):
+            volunteer_called = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                                   phone='+447519953189', camp=Camp(name='UCL'))
+            vc.edit_phone(volunteer=volunteer_called, phone='441111111111')
+
+    def test_edit_phone_too_short(self):
+        with self.assertRaises(vc.ControllerError):
+            volunteer_called = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                                   phone='+447519953189', camp=Camp(name='UCL'))
+            vc.edit_phone(volunteer=volunteer_called, phone='+44123')
+
+    def test_edit_camp_by_admin(self):
+        volunteer_called = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                               phone='+447519953189', camp=Camp(name='UCL'))
+        volunteer_edited = vc.edit_camp(volunteer=volunteer_called, camp=Camp(name='Euston'), is_admin=True)
+        self.assertEqual(str(volunteer_edited.camp), 'Euston')
+
+    def test_edit_camp_under_same_plan_by_volunteer(self):
+        camp_1 = Camp(name='Camp 1')
+        camp_2 = Camp(name='Camp 2')
+        Plan(name='Plan A',
+             emergency_type=Plan.EmergencyType.EARTHQUAKE,
+             description='Test',
+             geographical_area='',
+             camps=[camp_1, camp_2])
+        volunteer_called = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                               phone='+447519953189', camp=camp_1)
+        volunteer_edited = vc.edit_camp(volunteer=volunteer_called, camp=camp_2, is_admin=False)
+        self.assertEqual(str(volunteer_edited.camp), 'Camp 2')
+
+    def test_edit_camp_under_different_plan_by_volunteer(self):
+        with self.assertRaises(vc.ControllerError):
+            camp_1 = Camp(name='Camp 1')
+            camp_2 = Camp(name='Camp 2')
+            Plan(name='Plan A',
+                 emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                 description='Test',
+                 geographical_area='',
+                 camps=[camp_1])
+            Plan(name='Plan B',
+                 emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                 description='Test',
+                 geographical_area='',
+                 camps=[camp_2])
+            volunteer_called = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                                   phone='+447519953189', camp=camp_1)
+            vc.edit_camp(volunteer=volunteer_called, camp=camp_2, is_admin=False)
+
+    def test_edit_availability_false(self):
```

Comment:
Very minor point, but for consistency's sake with capitalisation in the next test this unit test should be 
```
def test_edit_availability_False(self):
```
Alternatively, the next test should have lower case true in title.

Reply:
Thanks for the reminder!